pkg/sources/github/github.go

```diff
@@ -772,50 +780,35 @@ func (s *Source) scan(ctx context.Context, installationClient *github.Client, ch
 				return nil
 			}
 
-			logger := s.log.WithValues("repo", repoURL)
-			logger.V(2).Info("attempting to clone repo")
-			var (
-				path string
-				repo *gogit.Repository
-				err  error
-			)
-
-			path, repo, err = s.cloneRepo(ctx, repoURL, installationClient)
+			// Scan the repository
+			repoCtx := context.WithValues(ctx, "repo", repoURL)
+			duration, err := s.cloneAndScanRepo(repoCtx, installationClient, repoURL, chunksChan)
 			if err != nil {
 				scanErrs.Add(err)
 				return nil
 			}
-
-			defer os.RemoveAll(path)
-			if err != nil {
-				scanErrs.Add(fmt.Errorf("error cloning repo %s: %w", repoURL, err))
-				return nil
-			}
-
-			s.setScanOptions(s.conn.Base, s.conn.Head)
-
-			repoSize := s.repoSizes.getRepo(repoURL)
-			logger.V(2).Info("scanning repo", "repo_size_kb", repoSize)
-
-			now := time.Now()
-			scanFailed := false
-			defer func(start time.Time, failed *bool) {
-				if *failed {
-					return
+			ctx.Logger().V(2).Info(fmt.Sprintf("scanned %d/%d repos", scannedCount, len(s.repos)), "duration_seconds", duration)
+
+			// Scan the wiki, if enabled, and the repo has one.
+			if s.conn.IncludeWikis {
+				if _, ok := s.reposWithWikis[repoURL]; ok {
+					wikiURL := strings.TrimSuffix(repoURL, ".git") + ".wiki.git"
+					wikiCtx := context.WithValue(ctx, "repo", wikiURL)
+
+					_, err := s.cloneAndScanRepo(wikiCtx, installationClient, wikiURL, chunksChan)
+					if err != nil {
+						scanErrs.Add(err)
+						// Don't return, it still might be possible to scan comments.
+					}
 				}
-				logger.V(2).Info(fmt.Sprintf("scanned %d/%d repos", scannedCount, len(s.repos)), "duration_seconds", time.Since(start).Seconds())
```

Comment:
Does this no longer happen if the scan panics? That's a fine change (and your version is more readable), I just want to ensure I understand the diff.

Reply:
> Does this no longer happen if the scan panics?

These changes were made based on how the code looked before #2220 was merged. Now that #2220 has been merged, the diff is a bit awkward.

Your understanding is correct. In fact, since @dustin-decker moved the metric and `scannedCount` increment to the end of the function, I should move this logging statement as well.